deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,146 @@
+        self.metric.reset()
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind.TRAIN):
+        """Update the metrics by passing the batch to ignite metric update method."""
+        label = batch.labels
+        prediction = batch.predictions
+        self.metric.update((prediction, label))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind.TRAIN) -> CheckResult:
+        """Compute the metric result using the ignite metrics compute method and reduce to a scalar."""
+        metric_result = self.metric.compute()
+        if self.reduce is not None:
+            if isinstance(metric_result, numbers.Real):
+                warnings.warn(SyntaxWarning('Metric result is already scalar, skipping reduce operation.'
+                                            'Pass reduce=None to prevent this'))
+                result_value = float(metric_result)
+            else:
+                result_value = float(self.reduce(metric_result))
+        elif isinstance(metric_result, float):
+            result_value = metric_result
+        else:
+            raise DeepchecksValueError(f'The metric {self.metric.__class__} return a non-scalar value, '
+                                       f'please specify a reduce function or choose a different metric')
+
+        result_dict = {'score': result_value,
+                       'metric': self.metric_name,
+                       'reduce': self.reduce_name}
+        return CheckResult(result_dict)
+
+    def add_condition_greater_than(self, threshold: float) -> ConditionResult:
+        """Add condition - the result is greater than the threshold."""
+        def condition(check_result):
+            details = f'The score {self.metric_name} is {check_result["score"]}'
+            if check_result['score'] > threshold:
+                return ConditionResult(ConditionCategory.PASS, details)
+            else:
+                return ConditionResult(ConditionCategory.FAIL, details)
+        return self.add_condition(f'Score is greater than {threshold}', condition)
+
+    def add_condition_greater_equal_to(self, threshold: float) -> ConditionResult:
+        """Add condition - the result is greater than the threshold."""
+        def condition(check_result):
+            details = f'The score {self.metric_name} is {check_result["score"]}'
+            if check_result['score'] >= threshold:
+                return ConditionResult(ConditionCategory.PASS, details)
+            else:
+                return ConditionResult(ConditionCategory.FAIL, details)
+        return self.add_condition(f'Score is greater than or equal to {threshold}', condition)
+
+    def add_condition_less_than(self, threshold: float) -> ConditionResult:
+        """Add condition - the result is greater than the threshold."""
+        def condition(check_result):
+            details = f'The score {self.metric_name} is {check_result["score"]}'
```

Comment:
are you sure inside the condition it have access to `self`?
maybe you need to do
```
def condition(check_result, metric_name=self.metric_name):
```

Reply:
It works...